### PR TITLE
Use public key thumbprint as KID in JWKS.json

### PR DESCRIPTION
**Pull request by cmiacz**
I want to merge this change because we have issues with key rotation.
It is expected that KID changes if we rotate RSA key for JWT. Right now we have hardcoded `KID: 1`

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

**Review by NyanKiyoshi — APPROVED** [review]

File: saleor/core/jwt_manager.py [hunk 1]
```diff
@@ -138,16 +142,25 @@ def get_public_key(cls) -> rsa.RSAPublicKey:
     @classmethod
     def get_jwks(cls) -> dict:
         jwk_dict = json.loads(RSAAlgorithm.to_jwk(cls.get_public_key()))
-        jwk_dict.update({"use": "sig", "kid": KID})
+        jwk_dict.update({"use": "sig", "kid": cls.get_key_id()})
         return {"keys": [jwk_dict]}
 
+    @classmethod
+    def get_key_id(cls) -> str:
```
Comment (NyanKiyoshi) [inline saleor/core/jwt_manager.py:149]:
```suggestion
    def get_key_id(cls) -> str:
        """Generate JWT key ID for the public key.
        
        This generates a "thumbprint" as 'kid' field using RFC 7638 implementation
        based on the RSA public key.
        """
```

**Review by NyanKiyoshi — APPROVED** [review]
Suggested a short docstring explaining what the class-method does and where to find extra information/docs (the RFC number)

**Review by korycins — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ All notable, unreleased changes to this project will be documented in this file.
 # 3.15.0 [Unreleased]
 
 ### Breaking changes
+- Use public key thumbprint as KID in JWKS.json #13442 by @cmiacz
```
Comment (maarcingebala) [inline CHANGELOG.md:9]: If this is a breaking change, we need to provide more context e.g. who is affected and how they should migrate to the new version. E.g. we can add a migration guide similar to this: https://docs.saleor.io/docs/3.x/upgrade-guides/3-13-to-3-14

**Comment by maarcingebala** [comment]
@cmiacz `black` is failing; the code needs to be formatted with it.

File: CHANGELOG.md (hunk 2 above)
Comment (cmiacz) [reply CHANGELOG.md]: It's not breaking chane, but fix for unexpected behavior. I've fixed the changelog file.

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**